src/synapse/azext_synapse/custom.py

```diff
@@ -0,0 +1,171 @@
+
+
+def cancel_spark_session_job(cmd, client, workspace_name, spark_pool_name, session_id):
+    return client.delete(workspace_name, spark_pool_name, session_id)
+
+
+def reset_timeout(cmd, client, workspace_name, spark_pool_name, session_id):
+    return client.reset_timeout(workspace_name, spark_pool_name, session_id)
+
+
+def list_spark_session_statements(cmd, client, workspace_name, spark_pool_name, session_id):
+    return client.list_statements(workspace_name, spark_pool_name, session_id)
+
+
+def create_spark_session_statement(cmd, client, workspace_name, spark_pool_name, session_id, code, kind):
+    livy_statement_request = LivyStatementRequestBody(code=code, kind=kind)
+    return client.create_statement(workspace_name, spark_pool_name, session_id, livy_statement_request)
+
+
+def get_spark_session_statement(cmd, client, workspace_name, spark_pool_name, session_id, statement_id):
+    return client.get_statement(workspace_name, spark_pool_name, session_id, statement_id)
+
+
+def cancel_spark_session_statement(cmd, client, workspace_name, spark_pool_name, session_id, statement_id):
+    return client.delete_statement(workspace_name, spark_pool_name, session_id, statement_id)
+
+
+def list_workspaces(cmd, client, resource_group_name=None):  # pylint: disable=unused-argument
+    return client.list_by_resource_group(
+        resource_group_name=resource_group_name) if resource_group_name else client.list()
+
+
+def create_workspace(cmd, client, resource_group_name, workspace_name, account_url, file_system, sql_admin_login_user,
+                     sql_admin_login_password, location, tags=None, identity_type="SystemAssigned"):
+    identity = ManagedIdentity(type=identity_type)
+    default_data_lake_storage = DataLakeStorageAccountDetails(account_url=account_url, filesystem=file_system)
+    workspace_info = Workspace(
+        identity=identity,
+        default_data_lake_storage=default_data_lake_storage,
+        sql_administrator_login=sql_admin_login_user,
+        sql_administrator_login_password=sql_admin_login_password,
+        location=location
+    )
+    return client.create_or_update(resource_group_name, workspace_name, workspace_info)
+
+
+def update_workspace(cmd, client, resource_group_name, workspace_name, sql_admin_login_password=None,
+                     provisioning_state=None, identity_type="SystemAssigned", principal_id=None, tags=None):
+    identity = ManagedIdentity(type=identity_type)
+    workspace_patch_info = WorkspacePatchInfo(
+        tags=tags,
+        identity=identity,
+        sql_administrator_login_password=sql_admin_login_password,
+        provisioning_state=provisioning_state
+    )
+    return client.update(resource_group_name, workspace_name, workspace_patch_info)
+
+
+def create_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name, location,
+                      spark_version="2.4", node_size=NodeSize.medium.value, node_count=3,

```

Comment:
>"2.4" [](start = 36, length = 5)

Strongly suggest not to provide default spark version. In 3 or 5 years, 2.4 might not be supported by RP. Then it will cause trouble to CLI users.

Reply:
removed default value

---
In reply to: [386420203](https://github.com/Azure/azure-cli-extensions/pull/1339#discussion_r386420203) [](ancestors = 386420203)